extensions/ql-vscode/src/remote-queries/shared/variant-analysis.ts

```diff
@@ -151,6 +151,14 @@ export function hasRepoScanCompleted(repo: VariantAnalysisScannedRepository): bo
   return isCompletedAnalysisRepoStatus(repo.analysisStatus);
 }
 
+/**
+ * @param repo
+ * @returns whether the repo scan is complete and has results that can be downloaded
```

Comment:
I don't think we're actually skipping the upload when there are no results, so we will always have results that can be downloaded. I think it's fine to skip the download, but I don't think we're skipping the download currently.

Reply:
Interesting. I agree we always upload an artefact, regardless of results. Skipping the download when there's no results makes sense to me as an optimisation, but I agree it's not strictly necessary.

For now I think I'd prefer to fit in with the existing behaviour, so that means we don't check if the number of results is non-zero. We can make that change later on if we want that as an optimisation, but I don't want us doing different behaviour in different places.

I also noticed that we could change `shouldDownload` from the `variant-analysis-monitor.ts` to call this method and that would let us show more logic.

Reply:
On trying to reuse the method it doesn't work because `shouldDownload` operates on a `gh_api` version of `VariantAnalysis`. I personally feel we should keep those API types as localised as possible and avoid them leaking around the extension, but I don't want to change this is this PR as it extends to the download command as well.

I'll leave this for now but open an issue so people can give their opinions there.